lib/matplotlib/cm.py

```diff
@@ -104,7 +146,8 @@ def register_cmap(name=None, cmap=None, data=None, lut=None):
             raise ValueError("Arguments must include a name or a "
                              "Colormap") from err
     if isinstance(cmap, colors.Colormap):
-        cmap_d[name] = cmap
+        cmap._global = True
```

Comment:
Please document in the docstring that if passing a Colormap instance, that instance will become immutable.

Reply:
I added in a Notes section to both the `get_cmap` and `register_cmap` functions now. Let me know if you want any different wording there. It may be odd to talk about future behavior there considering you can't do anything to get that future behavior yet? But I can't think of any great wording off the top of my head.